### PR TITLE
add enum support

### PR DESCRIPTION
**Pull request by gjo**

### CODE REVIEW

**Comment by tseaver** [comment]
Seems like a reasonable feature to me, and has tests / docs included.

**Comment by mmerickel** [comment]
I agree, very nice PR. Please add your name to contributors.txt.

File: colander/__init__.py [hunk 1]
```diff
@@ -1837,6 +1838,60 @@ def deserialize(self, node, cstruct):
...
     return datetime.datetime(*time.strptime(t, format)[0:6]).time()
 
+class Enum(SchemaType):
+    """A type representing a Python ``enum.Enum`` object.
+
+    The constructor accepts three arguments named ``enum_cls``, ``attr``
+    and ``typ``.
+
+    ``enum_cls`` is a mandatory argument and it should be a subclass of
+    ``enum.Enum``.  This argument represents appstruct's type.
```
Comment (stevepiercy) [inline colander/__init__.py]: Add missing "the":

```
represents the appstruct's
```

File: colander/__init__.py [hunk 2]
```diff
@@ -1837,6 +1838,60 @@ def deserialize(self, node, cstruct):
...
+
+    ``attr`` is a optional argument.  default is ``name``.
```
Comment (stevepiercy) [inline colander/__init__.py]: Use "an", not "a", before a noun beginning with a vowel. Also grammar fix:

```
``attr`` is an optional argument. Its default is ``name``.
```

File: colander/__init__.py [hunk 3]
```diff
@@ -1837,6 +1838,60 @@ def deserialize(self, node, cstruct):
...
+    It is used to pick a serialized value from a enum instance.
```
Comment (stevepiercy) [inline colander/__init__.py]:
```
an enum
```

File: colander/__init__.py [hunk 4]
```diff
@@ -1837,6 +1838,60 @@ def deserialize(self, node, cstruct):
...
+    A serialized value must be unique.
+
+    ``typ`` is a optional argument and it shoud be a instance of
```
Comment (stevepiercy) [inline colander/__init__.py]:
```
``typ`` is an optional argument, and it should be an instance of
```

File: colander/__init__.py [hunk 5]
```diff
@@ -1837,6 +1838,60 @@ def deserialize(self, node, cstruct):
...
+    ``colander.SchemaType``.  This argument represents cstruct's type.
```
Comment (stevepiercy) [inline colander/__init__.py]:
```
represents the cstruct's
```

**Comment by gjo** [comment]
I already signed to CONTRIBUTEORS.txt:124

**Comment by stevepiercy** [comment]
@tseaver @mmerickel should we add the `enum` type to the list of supported types on https://docs.pylonsproject.org/projects/colander/en/master/ to align with this PR?

**Pull request merged**